src/Raven.Server/Documents/Indexes/Static/V8/JavaScriptMapOperation.cs

```diff
@@ -0,0 +1,143 @@
+                        catch (V8Exception jse)
+                        {
+                            var (message, success) = JavaScriptIndexFuncException.PrepareErrorMessageForJavaScriptIndexFuncException(MapString, jse);
+                            jsRes.Dispose();
+                            if (success == false)
+                                throw new JavaScriptIndexFuncException($"Failed to execute {MapString}", jse);
+                            throw new JavaScriptIndexFuncException($"Failed to execute map script, {message}", jse);
+                        }
+                        catch (Exception e)
+                        {
+                            jsRes.Dispose();
+                            throw new JavaScriptIndexFuncException($"Failed to execute {MapString}", e);
+                        }
+                        finally
+                        {
+                            _engine.ForceV8GarbageCollection();
+                        }
+
+                        using (jsRes)
+                        {
+                            if (jsRes.IsArray)
+                            {
+                                var length = (uint)jsRes.ArrayLength;
+                                for (int i = 0; i < length; i++)
+                                {
+                                    var arrItem = jsRes.GetProperty(i);
+                                    using (arrItem) { 
+                                        if (arrItem.IsObject)
+                                        {
+                                            yield return arrItem; // being yield it is converted to blittable object and not disposed - so disposing it here
+                                        }
+                                        else
+                                        {
+                                            // this check should be to catch map errors
+                                            throw new JavaScriptIndexFuncException($"Failed to execute {MapString}", new Exception($"At least one of map results is not object: {jsRes.ToString()}"));
+                                        }
+                                    }
+                                }
+                            }
+                            else if (jsRes.IsObject)
+                            {
+                                yield return jsRes;// being yield it is converted to blittable object and not disposed - so disposing it here
+                            }
+                            else
+                            {
+                                // this check should be to catch map errors
+                                throw new JavaScriptIndexFuncException($"Failed to execute {MapString}", new Exception($"Map result is not object: {jsRes.ToString()}"));
+                            }
+                        }
+                        // we ignore everything else by design, we support only
```

Comment:
Didn't we add an error here recently?

Reply:
for backward compatibility I have removed throwing the error in case map returns neither array or object

Reply:
as now we have the single version of JavaScriptMapOperation